python3Packages.ttp: FileNotFoundError: template.extend: failed load extend tag content at path: ./assets/test_extend_tag_bgp_config.txt

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `python3Packages.ttp`

### Build log
```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/bzd527vinla7nd73ralrsz28njnd7p62-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/ttp/variable/time_funcs.py
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing pipBuildPhase
Creating a wheel...
[33mWARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.[0m[33m
[0mProcessing /build/source
  Running command Preparing metadata (pyproject.toml)
  Preparing metadata (pyproject.toml) ... [?25l[?25hdone
Building wheels for collected packages: ttp
  Running command Building wheel for ttp (pyproject.toml)
  Building wheel for ttp (pyproject.toml) ... [?25l[?25hdone
  Created wheel for ttp: filename=ttp-0.9.1-py2.py3-none-any.whl size=84602 sha256=d12cc6427a2d91c0718a6a5e4d8c684b442645b43d58f2790c605154824507f3
  Stored in directory: /build/pip-ephem-wheel-cache-g33je566/wheels/79/4b/9b/fc24941d75fdb959a71ee80195b2c89d6ae9389f9f0f80fb22
Successfully built ttp
Finished creating a wheel...
Finished executing pipBuildPhase
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./ttp-0.9.1-py2.py3-none-any.whl
Installing collected packages: ttp
Successfully installed ttp-0.9.1
/build/source
Finished executing pipInstallPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/ig4zwszs35zhl6g7lim5sfpmk8iklpr2-python3.10-ttp-0.9.2
checking for references to /build/ in /nix/store/ig4zwszs35zhl6g7lim5sfpmk8iklpr2-python3.10-ttp-0.9.2...
patching script interpreter paths in /nix/store/ig4zwszs35zhl6g7lim5sfpmk8iklpr2-python3.10-ttp-0.9.2
strip is /nix/store/iiq295j1z4q1sxmdbrl2j8ma3l5ns4wr-gcc-wrapper-11.3.0/bin/strip
stripping (with command strip and flags -S) in  /nix/store/ig4zwszs35zhl6g7lim5sfpmk8iklpr2-python3.10-ttp-0.9.2/lib /nix/store/ig4zwszs35zhl6g7lim5sfpmk8iklpr2-python3.10-ttp-0.9.2/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/b1vwvzfca0g6cis1g101zs9kbhwji1c1-python3.10-ttp-0.9.2-dist
checking for references to /build/ in /nix/store/b1vwvzfca0g6cis1g101zs9kbhwji1c1-python3.10-ttp-0.9.2-dist...
patching script interpreter paths in /nix/store/b1vwvzfca0g6cis1g101zs9kbhwji1c1-python3.10-ttp-0.9.2-dist
strip is /nix/store/iiq295j1z4q1sxmdbrl2j8ma3l5ns4wr-gcc-wrapper-11.3.0/bin/strip
Rewriting #!/nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/bin/python3.10 to #!/nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9
wrapping `/nix/store/ig4zwszs35zhl6g7lim5sfpmk8iklpr2-python3.10-ttp-0.9.2/bin/ttp'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
@nix { "action": "setPhase", "phase": "installCheckPhase" }
running install tests
no Makefile or custom installCheckPhase, doing nothing
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: ttp
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
pytestCheckPhase
Executing pytestCheckPhase
[1m============================= test session starts ==============================[0m
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
rootdir: /build/source
[1mcollecting ... [0m[1m
collecting 187 items                                                           [0m[1m
collected 252 items / 23 deselected / 229 selected                             [0m

test/pytest/test_anonymous_group.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                 [  1%][0m
test/pytest/test_answers_and_docs.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m [ 17%]
[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33ms[0m[32m.[0m[33ms[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                        [ 31%][0m
test/pytest/test_empty_group_results.py [32m.[0m[33m                                [ 31%][0m
test/pytest/test_group_chain_attribute.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                           [ 33%][0m
test/pytest/test_group_functions.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                   [ 41%][0m
test/pytest/test_group_lookup.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                 [ 44%][0m
test/pytest/test_group_name_attribute.py [32m.[0m[33m                               [ 44%][0m
test/pytest/test_headers_indicator.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                        [ 49%][0m
test/pytest/test_inputs.py [32m.[0m[33m                                             [ 50%][0m
test/pytest/test_line_indicator.py [32m.[0m[33m                                     [ 50%][0m
test/pytest/test_match_condition_fun.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m            [ 59%][0m
test/pytest/test_match_functions.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                       [ 65%][0m
test/pytest/test_misc.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                   [ 71%][0m
test/pytest/test_output_formatters.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                       [ 76%][0m
test/pytest/test_output_functions.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                              [ 79%][0m
test/pytest/test_output_returners.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                               [ 81%][0m
test/pytest/test_regexes.py [32m.[0m[32m.[0m[32m.[0m[33m                                          [ 82%][0m
test/pytest/test_structure.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                     [ 85%][0m
test/pytest/test_syslog_returner.py [32m.[0m[33m                                    [ 86%][0m
test/pytest/test_template_variables.py [32m.[0m[33m                                 [ 86%][0m
test/pytest/test_ttp_parser_methods.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[31m   [100%][0m

=================================== FAILURES ===================================
[31m[1m__________________ test_get_template_method_nested_templates ___________________[0m

    def test_get_template_method_nested_templates():
        template = """
    <macro>
    def indent1(template_text):
        return "\\n".join(f" {line}" for line in template_text.splitlines())
    
    def indent2(template_text):
        return "\\n".join(f"  {line}" for line in template_text.splitlines())
    </macro>
    
    <group name="bgp">
    router bgp {{ asn }}
     <extend template="./assets/test_extend_tag_bgp_config.txt" macro="indent1"/>
    
    
     <group name="vrfs*">
     vrf {{ vrf }}
      rd {{ rd }}
      <extend template="./assets/test_extend_tag_bgp_config.txt" macro="indent2"/>
     </group>
    
    </group>
    
    <extend template="./assets/extend_vlan.txt"/>
    
    <group>
    interface {{ interface }}
     description {{ description | ORPHRASE }}
     ip address {{ ip }} {{ mask }}
    </group>
        """
>       parser = ttp(template=template)

[1m[31mtest/pytest/test_ttp_parser_methods.py[0m:1854: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[1m[31mttp/ttp.py[0m:246: in __init__
    self.add_template(template=template)
[1m[31mttp/ttp.py[0m:368: in add_template
    template_obj = _template_class(
[1m[31mttp/ttp.py[0m:948: in __init__
    self.template = self.handle_extend(template_text)
[1m[31mttp/ttp.py[0m:1368: in handle_extend
    self.handle_extend(template_ET=child, top=False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ttp.ttp._template_class object at 0x7fffef8fd4e0>, template_text = None
template_ET = <Element 'group' at 0x7fffef8e6a70>, top = False

    def handle_extend(self, template_text=None, template_ET=None, top=True):
        """
        Function to handle all extend tags including nested within groups.
    
        :param template_text: (str) template string
        :return: (str) final template string after extending it
        """
        # load template string to XML etree object
        if template_text:
            template_ET = self.construct_etree(template_text)
        # load macro functions from top template into self.macro dict
        if top == True:
            for child in template_ET:
                if child.tag == "macro":
                    self.parse_macro(child)
        # recursively iterate over etree handling extend tags
        for index, child in enumerate(template_ET):
            if child.tag == "extend":
                path_to_template = child.attrib.get("template", "")
                content = self._ttp_["utils"]["load_files"](path_to_template, read=True)
                if content[0][1] == path_to_template:
>                   raise FileNotFoundError(
                        "template.extend: failed load extend tag content at path: {}".format(
                            path_to_template
                        )
                    )
[1m[31mE                   FileNotFoundError: template.extend: failed load extend tag content at path: ./assets/test_extend_tag_bgp_config.txt[0m

[1m[31mttp/ttp.py[0m:1338: FileNotFoundError
[33m=============================== warnings summary ===============================[0m
test/pytest/test_answers_and_docs.py:6747
  /build/source/test/pytest/test_answers_and_docs.py:6747: DeprecationWarning: invalid escape sequence '\s'
    template = """

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[36m[1m=========================== short test summary info ============================[0m
[31mFAILED[0m test/pytest/test_ttp_parser_methods.py::[1mtest_get_template_method_nested_templates[0m - FileNotFoundError: template.extend: failed load extend tag content at path:...
[31m====== [31m[1m1 failed[0m, [32m226 passed[0m, [33m2 skipped[0m, [33m23 deselected[0m, [33m1 warning[0m[31m in 2.79s[0m[31m ======[0m
/nix/store/chilfhdcsnmwjl7igrw26j1lrc0zar35-stdenv-linux/setup: line 1570: pop_var_context: head of shell_variables not a function context

```

### Additional context

I started encountering this error when I tried to upgrade to the latest nixos-unstable-small c85d08692966cf022b0a741a794cb1650602d8af, because `ansible` depends on `ttp`.

I see `"python3Packages.ttp:` was upgraded recently in 8867ba18082bf29d2a7854df79937a54a7b507dc, and reverting that commit does fix the build for me, but it may make more sense to fix the package directly.

That got merged to master via staging-next  in https://github.com/NixOS/nixpkgs/pull/209180

### Notify maintainers

@mweinelt 


### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.17, NixOS, 23.05 (Stoat), 23.05.git.befc83905c96`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos-21.11pre329341.9e70e9f732f"`
 - nixpkgs: `/etc/nixos/aij/unstable`

```

--- a/pkgs/development/python-modules/ttp/default.nix
+++ b/pkgs/development/python-modules/ttp/default.nix
@@ -62,6 +62,7 @@ buildPythonPackage rec {
     "test/pytest/test_N2G_formatter.py"
     # missing test file
     "test/pytest/test_extend_tag.py"
+    "test/pytest/test_ttp_parser_methods.py"
   ];
 
   disabledTests = [

